src/init.rs

```diff
@@ -11,29 +12,46 @@ rest of the script, and you have to spam semicolons everywhere. By using
 source and process substitutions, we make it possible to comment and debug
 the init scripts. */
 
+fn get_starship_path() -> io::Result<String> {
```

Comment:
Left a long-winded comment about this and potential privilege escalation attacks on the PR. Long and short, I'm not sure this is secure and I don't think it's salvageable with `current_exe`, at least on Linux (where the implementation reads /proc/self/exe`.

Do you think it would be possible to do this by reading `argv[0]` and using `fs::canonicalize`? That should get us a pretty similar effect without worrying about hardlink trickery. The only problem is that I'm not sure how to read `argv[0]` using clap (there was an issue opened about it, but it seems to have died without resolution).

Reply:
I did a quick test in `main()`.

```rust
use std::env;
println!("current_exe: {:?} {:?}", std::env::current_exe(), std::env::current_exe().map(|v| std::fs::canonicalize(v)));

for argument in env::args() {
    println!("ARG> {} {:?}", argument.clone(), std::fs::canonicalize(argument));
}
```

I created an hardl link (test_hardlink) pointing to starship and I got:
```
current_exe: Ok("/home/bbigras/test_hardlink") Ok(Ok("/home/bbigras/test_hardlink"))
ARG> ./test_hardlink Ok("/home/bbigras/test_hardlink")
thread 'main' panicked at 'not yet implemented', src/main.rs:28:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Reply:
I've gotten a few opinions from people who are better at systems stuff than me. I think I'm okay with using current_exe for this--the danger comes when using an suid binary that relaunches itself.

Reply:
Excellent, thanks!